### PR TITLE
Upgrade nameservice contract to cosmwasm 0.7.0

### LINKED ISSUE #42: Upgrade all contracts to cosmwasm 0.7
This involves adjusting to all breaking changes. But also making use of enhancements, like Base64/Binary for better representation of binary data in json (see #39)

### PR DESCRIPTION
**Pull request by webmaster128**
Version 0.1.0 was published to https://crates.io/crates/cw-nameservice. I'll publish 0.2.0 as soon as this is merged.

Closes #42

### CODE REVIEW

File: nameservice/src/contract.rs [hunk 1]
```diff
@@ -61,23 +61,27 @@ pub fn try_register<S: Storage, A: Api>(
...
+            if env.message.signer != record.owner {
+                unauthorized()?;
+            }
 
-        record.owner = new_owner.clone();
-        Ok(record)
+            record.owner = new_owner.clone();
+            Ok(record)
+        } else {
+            unauthorized()
```
Comment (ethanfrey) [inline nameservice/src/contract.rs]: nitpick: I would return not_found here (difference from not being able to transfer someone else's name, and the name not being registered at all)

Comment (ethanfrey) [reply nameservice/src/contract.rs]: (This was also the old behavior, implicit when calling `load` not `may_load` inside `cw-storage`)

Comment (webmaster128) [reply nameservice/src/contract.rs]: Thanks, will update! I was not sure there and did the migration somewhat mechanically.

Comment (webmaster128) [reply nameservice/src/contract.rs]: NotFound does not have a helper function and I think the `kind` field with a type is not very helpful. I used a `contract_err` now (74d79f5e9c4d21c53f4367bacc46f7b5a21a7564)

**Comment by webmaster128** [comment]
I'm rebuilding the contract.wasm right now

**Comment by ethanfrey** [comment]
Closes #42 right?

**Comment by webmaster128** [comment]
Right. Done from my side. Good to merge?

**Pull request merged**